aspects/aspects.go

```diff
@@ -310,78 +310,143 @@ func (a *Aspect) Set(databag DataBag, name string, value interface{}) error {
 		Account:    a.bundle.Account,
 		BundleName: a.bundle.Name,
 		Aspect:     a.Name,
-		Field:      name,
+		Field:      request,
 		Cause:      "field not found",
 	}
 }
 
-// Get returns the aspect value identified by the name. If either the named aspect
-// or the corresponding value can't be found, a NotFoundError is returned.
-func (a *Aspect) Get(databag DataBag, name string, value interface{}) error {
-	subkeys := strings.Split(name, ".")
-	for _, accessPatt := range a.accessPatterns {
-		placeholders, ok := accessPatt.match(subkeys)
-		if !ok {
-			continue
-		}
-
-		path, err := accessPatt.getPath(placeholders)
-		if err != nil {
-			return err
-		}
-
-		if !accessPatt.isReadable() {
-			return &InvalidAccessError{RequestedAccess: read, FieldAccess: accessPatt.access, Field: name}
-		}
+// Get returns the aspect value identified by the request. If either the named
+// aspect or the corresponding value can't be found, a NotFoundError is returned.
+func (a *Aspect) Get(databag DataBag, request string, value interface{}) error {
+	matches, err := a.matchGetRequest(request)
+	if err != nil {
+		return err
+	}
 
-		if err := databag.Get(path, value); err != nil {
+	var results []interface{}
+	for _, match := range matches {
```

Comment:
I think that makes sense when the request side is also fully contained. If the requests are different and they map to related storage values, returning only one would look like data is missing, from the user's perspective. Bit of an awkward example but:
```
  request: snaps.health
  storage: snaps.snapd.status

  request: snaps.all
  storage: snaps
```
If I do a Get with "snaps", I would expect it to match with both entries and get both results since from the perspective of the user they're different things. If we only consider the storage, we'd only get the second one. The data used in the first is included in the second but as a user I wouldn't know that since there would be an abstraction over the storage layer.

Reply:
If that's ok, I can make a ticket to do this as a follow-up